### PR TITLE
Model updates

### PR DESCRIPTION
**Pull request by dgasmith**
Forbids extras on all base models, and updates fields correspondingly.

### CODE REVIEW

File: qcelemental/models/common_models.py [hunk 1]
```diff
@@ -1,55 +1,58 @@
...
 
 ndarray_encoder = {np.ndarray: lambda v: v.flatten().tolist()}
 
 
 class Provenance(BaseModel):
     creator: str
-    version: str = None
-    routine: str = None
+    version: Optional[str] = None
+    routine: Optional[str] = None
```
Comment (loriab) [inline qcelemental/models/common_models.py]: this means that all three of creator/version/routine must be present in an instance but latter two may be defaulted if not supplied? molparse is demanding all three be supplied.

File: qcelemental/models/common_models.py [hunk 2]
```diff
@@ -1,55 +1,58 @@
...
 
     class Config:
-        extra = Extra.allow
+        extra = "allow"
 
 
 class Model(BaseModel):
     method: str
-    basis: str = None
+    basis: Optional[str] = None
```
Comment (loriab) [inline qcelemental/models/common_models.py]: this'll be helpful for d3

File: qcelemental/models/common_models.py (hunk 1 above)
Comment (dgasmith) [reply qcelemental/models/common_models.py]: Yes, this was always the case. The `Optional` isn't actually used unless a `mypy --strict-optional` is set. In general, version and routine are always set.

Comment (loriab) [reply qcelemental/models/common_models.py]: ok. so long as we don't hit cases that pass you and fail me, all's well.

**Pull request merged**